MinimedKit/PumpEvents/JournalEntryInsulinMarkerPumpEvent.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  JournalEntryInsulinMarkerPumpEvent.swift
+//  RileyLink
+//
+//  Created by Pete Schwamb on 7/16/16.
+//  Copyright © 2016 Pete Schwamb. All rights reserved.
+//
+
+import Foundation
+
+public struct JournalEntryInsulinMarkerPumpEvent: TimestampedPumpEvent {
+    public let length: Int
+    public let rawData: NSData
+    public let timestamp: NSDateComponents
+    public let amount: Double
+    
+    public init?(availableData: NSData, pumpModel: PumpModel) {
+        length = 8
+        
+        guard length <= availableData.length else {
+            return nil
+        }
+        
+        rawData = availableData[0..<length]
+        
+        timestamp = NSDateComponents(pumpEventData: availableData, offset: 2)
+        
+        let lowBits = rawData[1] as UInt8
+        let highBits = rawData[4] as UInt8
+        amount = Double((Int(highBits & 0b1100000) << 3) + Int(lowBits)) / 10.0
```

Comment:
Right, I assumed you'd apply the mask and combine into an array, but I'm not sure how much code that actually saves :)

> On Jul 16, 2016, at 10:13 PM, Pete Schwamb <notifications@github.com> wrote:
> 
> In MinimedKit/PumpEvents/JournalEntryInsulinMarkerPumpEvent.swift:
> 
> > +    public let amount: Double
> > +    
> > +    public init?(availableData: NSData, pumpModel: PumpModel) {
> > +        length = 8
> > +        
> > +        guard length <= availableData.length else {
> > +            return nil
> > +        }
> > +        
> > +        rawData = availableData[0..<length]
> > +        
> > +        timestamp = NSDateComponents(pumpEventData: availableData, offset: 2)
> > +        
> > +        let lowBits = rawData[1] as UInt8
> > +        let highBits = rawData[4] as UInt8
> > +        amount = Double((Int(highBits & 0b1100000) << 3) + Int(lowBits)) / 10.0
> No, this isn't contiguous. (byte 1 has low bits and byte 4 has high)
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>